testing/regress/hpcc/regression/cleanup.py

```diff
@@ -0,0 +1,101 @@
+logger = logging.getLogger('cleanup') 
+
+# Custom logger formatter for serial numbering of log records
+class SerialNumberFormatter(logging.Formatter):
+    def __init__(self):
+        super().__init__()
+        self.serialNumber = 0
+
+    def format(self, record):
+        self.serialNumber += 1
+        record.msg = str(self.serialNumber) + ". " + record.msg
+        return super().format(record)
+
+# Builds and configures the cleanup logger
+def buildCleanupLogger(logDir, cleanupLogger):
+    cleanupLogger.setLevel(logging.INFO)
+    curTime = time.strftime('%y-%m-%d-%H-%M-%S')
+    logName = "cleanup" + "." + curTime + ".log"
+    logPath = os.path.join(logDir, logName)
+    cleanupHandler = logging.FileHandler(logPath)
+    cleanupHandler.setFormatter(SerialNumberFormatter())
+    cleanupLogger.addHandler(cleanupHandler)
+    return cleanupLogger
+
+# Loads configuration data from JSON file
+def loadConfig(filename):
+    with open(filename, 'r') as file:
+        return json.load(file)
+
+# Extracts regress log files corresponding to current run of RTE
+def getRegressLogs(mode, logDirPath, startTime, endTime):
+    regressLogFiles = glob.glob(os.path.join(logDirPath, '*.log'))   
+    for logFilePath in regressLogFiles:
+        logFileNamePattern = r'^(thor|roxie|hthor)\.\d{2}-\d{2}-\d{2}-\d{2}-\d{2}-\d{2}\.log(?!.*exclusion)'
+        match = re.search(logFileNamePattern, os.path.basename(logFilePath))
+        baseFilePath = os.path.basename(logFilePath)
+        if match:
+            pathParts = baseFilePath.split('.') 
+            fileDatetimeStr = pathParts[1]
+            dateFormat = "%y-%m-%d-%H-%M-%S"
+            fileDate = datetime.strptime(fileDatetimeStr, dateFormat)
+            startDate = datetime.strptime(startTime, dateFormat)
+            endDate = datetime.strptime(endTime, dateFormat)
+            if startDate <= fileDate <= endDate:
+                getWorkunitDetails(logFilePath, mode)
+
+# Extracts workunit details from log file based on the cleanup mode
+def getWorkunitDetails(logFilePath, mode):
+    with open(logFilePath, 'r') as file:
+        lineList = file.readlines()
+        if mode == "all" or mode == "workunits":
+            for line in lineList:
+                index = line.find("URL")
+                if index != -1:
+                    url = line[index + 3:-1]
+                    deleteWorkunit(url)
+        elif mode == "passed":
+            for i, line in enumerate(lineList):
+                if re.search(r'\bpass\b', line, re.IGNORECASE):
+                    nextLine = lineList[i + 1]
+                    index = nextLine.find("URL")
+                    if index != -1:
+                        url = nextLine[index + 3:-1]
+                        deleteWorkunit(url)
+
+# Deletes workunits
+def deleteWorkunit(url):
+    deletionUrl = url.replace("?Widget=WUDetailsWidget&Wuid", "WsWorkunits/WUDelete.json?Wuids")
+    startIndex = url.find('Wuid=')
+    if startIndex != -1:
```

Comment:
What is happening if  (somehow) 'startIndex == -1'? 
Can it happen? If yes, then you should handle it. If not why this check

Reply:
Thank you for pointing out, we understand that once url has been found, it would definitely contain the 'Wuid=' string. Hence, the if condition is not necessary.